zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/DynamicCredentialsFileLoader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.elasticsearch;
+
+import com.google.common.util.concurrent.RateLimiter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.PASSWORD_PROP;
+import static zipkin2.server.internal.elasticsearch.ZipkinElasticsearchStorageConfiguration.USERNAME_PROP;
+
+/**
+ * Loads username/password from credentials file.
+ *
+ * <p><em>NOTE:</em> This implementation currently uses blocking I/O loop without poll events.
```

Comment:
I don't think we need this note. This is using blocking I/O on a separate thread from the server event loop so shouldn't really effect the API.

Reply:
the note is mostly about "without poll events" meaning there's no file change notification. I agree the "blocking I/O" word should be removed as it confuses.. I meant to say "not nio"? like old school file apis.

Reply:
Ah this is if the creds become invalidated in the middle of an interval, not about event loop. Yeah blocking tricked me. `Because this doesn't use FileWatcher` is probably clear to any Java developer